tests/JoinSqlTest.php

```diff
@@ -664,4 +664,87 @@ public function testJoinReverseOneOnOne()
         $this->assertTrue($m->loaded());
         $this->assertEquals(['id' => 23, 'name' => 'Chris', 'notes' => '5th note'], $m->get());
     }
+
+    public function testJoinActualFieldNamesAndPrefix()
+    {
+        if ($this->getDatabasePlatform() instanceof PostgreSQL94Platform || $this->getDatabasePlatform() instanceof SQLServer2012Platform || $this->getDatabasePlatform() instanceof OraclePlatform) {
+            $this->markTestIncomplete('TODO - NULL PK not unset in INSERT');
+        }
+
+        $this->setDb([
+            'user' => [
+                1 => ['id' => 1, 'first_name' => 'John', 'cid' => 1],
+                2 => ['id' => 2, 'first_name' => 'Peter', 'cid' => 1],
+                3 => ['id' => 3, 'first_name' => 'Joe', 'cid' => 2],
+            ], 'contact' => [
+                1 => ['id' => 1, 'contact_phone' => '+123'],
+                2 => ['id' => 2, 'contact_phone' => '+321'],
+            ], 'salaries' => [
+                1 => ['id' => 1, 'amount' => 123, 'uid' => 1],
+                2 => ['id' => 2, 'amount' => 456, 'uid' => 2],
+            ],
+        ]);
+
+        $db = new Persistence\Sql($this->db->connection);
+        $m_u = new Model($db, ['table' => 'user']);
+        $m_u->addField('contact_id', ['actual' => 'cid']);
+        $m_u->addField('name', ['actual' => 'first_name']);
+        // normal join
+        $j = $m_u->join('contact', ['prefix' => 'j1_']);
+        $j->addField('phone', ['actual' => 'contact_phone']);
+        // reverse join
+        $j2 = $m_u->join('salaries.uid', ['prefix' => 'j2_']);
+        $j2->addField('salary', ['actual' => 'amount']);
+
+        // update
+        $m_u2 = (clone $m_u)->load(1);
+        $m_u2->set('name', 'John 2');
+        $m_u2->set('j1_phone', '+555');
+        $m_u2->set('j2_salary', 111);
+        $m_u2->save();
+
+        $this->assertEquals(
+            [
+                'user' => [
+                    1 => ['id' => 1, 'first_name' => 'John 2', 'cid' => 1],
```

Comment:
I belive this is wrong, as `actual` name should be used for low level SQL only, otherwise field name (not `actual` name) should be used.

Reply:
Not really. This is just direct dump from DB, so there will be actual names here.

Reply:
You are right, but in export and other non-low-level dumps it should not have the `actual` name.

Reply:
yes in import / export should not have actual names.

But as you see in Join class it still will use actual names when linking tables. See I added comment there in this PR.
I don't think we have to change that, but having additional note about that could be good.